src/main/java/com/hubspot/jinjava/el/ext/JinjavaBeanELResolver.java

```diff
@@ -126,17 +128,40 @@ private boolean isRestrictedClass(Object o) {
       return false;
     }
 
-    return (
-      (
-        o.getClass().getPackage() != null &&
-        o.getClass().getPackage().getName().startsWith("java.lang.reflect")
-      ) ||
+    if (
       o instanceof Class ||
       o instanceof ClassLoader ||
       o instanceof Thread ||
       o instanceof Method ||
       o instanceof Field ||
-      o instanceof Constructor
-    );
+      o instanceof Constructor ||
+      o instanceof JinjavaInterpreter
+    ) {
+      return true;
+    }
+
+    if (o.getClass().getPackage() != null) {
+      if (o.getClass().getPackage().getName().startsWith("java.lang.reflect")) {
+        return true;
+      } else if (o.getClass().getPackage().getName().startsWith("com.hubspot.jinjava")) {
+        return false;
+      }
+    }
+
+    JinjavaInterpreter interpreter = JinjavaInterpreter.getCurrent();
+    if (interpreter != null && interpreter.getConfig() != null) {
+      if (!interpreter.getConfig().getAllowedHostClasses().contains(o.getClass())) {
+        if (interpreter.getConfig().getEnforceHostWhitelist()) {
+          return true;
+        } else {
+          Logging.ENGINE_LOG.error(
+            "Warning: Template using method on restricted class '{}'",
```

Comment:
Not necessarily a template.

Reply:
Isn't everything technically a template to Jinjava? Trying to keep this copy specific to Jinjava rather than HubSpot.